docs/admin.rst

```diff
@@ -1,6 +1,71 @@
+the appropriate Treeherder model classes and generating Django
+schema migrations. However, there are some MySQL-specific gotchas
+which apply when performing these operations on larger tables (i.e.
+those with millions of rows or more).
+
+* Django's schema migrations are often rather inefficient for the
+  case where you are modifying multiple columns on a table at the
+  same time (it will generate as many alter table statements as
+  there are changes, when they could all be consolidated into one)
+* Altering a column with a foreign key relationship (i.e. changing
+  its type or adding/removing a null constraint) will cause MySQL
+  to re-check the validity of that relation for every row, which
+  can be quite expensive.
+* `It does not seem to be possible <https://dev.mysql.com/doc/refman/5.6/en/innodb-create-index-limitations.html>`_ to add or alter a table column
+  while additional rows are being added to the table without
+  creating duplicates (which will then cause the whole operation
+  to fail due to integrity errors)
+
+For this reason, it is recommended that you bypass using Django
+migrations and follow this procedure when modifying any large table:
+
+* Generate the SQL in a vagrant instance for the schema changes
+  you want to make (`./manage.py sqlmigrate model <name of migration>`
+  will print this out to the console) and modify if needed (i.e.
+  consolidating `ALTER TABLE` statements). For auditing purposes,
+  please post this SQL to an attachment on the bug (it may also
+  be a good idea to have another Treeherder dev look it over).
+* Temporarily turn off job ingestion (you can do this by changing
+  the number of dynos devoted to job ingestion in Heroku's dashboard).
+  Ideally it would be best to do this in a period of slightly lower activity.
+* Open a SQL session, `SET FOREIGN_KEY_CHECKS=0;` (this is a
+  session-specific option, so no need to worry about resetting it
+  later), then apply your changes.
+* When the SQL is fully applied, manually generate an entry in the
+  django_migrations table as follows: `insert into django_migrations
+  (app, name, applied) values (<app>, <migration name>,
+  <date>)`. Select a few rows from the table to see the
+  expected format. For date, you can use the 'YYYY-MM-DD' format.
+* Re-enable job ingestion.
+* After this is done, you should be able to actually deploy your
+  code change to Treeherder.
+
+As an alternative to disabling job ingestion (which can be rather
+intrusive), it is sometimes possible to perform a schema change on
+a *copy* of your table, import the data into it, then swap the
+tables.
+
+For example:
+
+.. code-block:: sql
+
+    CREATE table performance_datum_copy LIKE performance_datum;
+    <schema changes on performance_datum_copy>
+    INSERT INTO performance_datum_copy SELECT * FROM performance_datum;
+    SET @max_perf_id = (select max(id) FROM performance_datum_copy);
+    INSERT INTO performance_datum_copy SELECT * FROM performance_datum
```

Comment:
There is no guarantee, but I think in practice the window of time is small enough that I would not expect any issues. We could probably use a table lock here if you were really worried.